webui/react/src/services/api.ts

```diff
@@ -13,19 +13,34 @@ import {
   DetailedUser, DeterminedInfo, ExperimentBase, ExperimentDetails,
   ExperimentFilters, ExperimentItem, Log, Pagination, RunState, TrialDetails,
 } from 'types';
+import { globalStorage, STORAGE_KEY_AUTH } from 'utils/storage';
 import { isExperimentTask } from 'utils/task';
 import { terminalCommandStates, tsbMatchesSource } from 'utils/types';
 
 import { decodeExperimentList, encodeExperimentState } from './decoder';
 
 export { isAuthFailure, isLoginFailure, isNotFound } from './utils';
 
-const ApiConfig : Api.Configuration = { basePath: serverAddress() };
-
+const initialApiConfig = {
+  apiKey: 'Bearer ' + globalStorage.get(STORAGE_KEY_AUTH),
+  basePath: serverAddress(),
+};
 export const detApi = {
-  Auth: new Api.AuthenticationApi(ApiConfig),
-  Experiments: new Api.ExperimentsApi(ApiConfig),
-  StreamingExperiments: Api.ExperimentsApiFetchParamCreator(),
+  Auth: new Api.AuthenticationApi(initialApiConfig),
+  Experiments: new Api.ExperimentsApi(initialApiConfig),
+  StreamingExperiments: Api.ExperimentsApiFetchParamCreator(initialApiConfig),
+};
+
+// Update references to generated API code with new configuration.
+export const updateDetApi = (apiConfig: Api.Configuration): void => {
+  apiConfig = {
+    apiKey: 'Bearer ' + globalStorage.get(STORAGE_KEY_AUTH),
+    basePath: serverAddress(),
+    ...apiConfig,
+  };
+  detApi.Auth = new Api.AuthenticationApi(apiConfig);
+  detApi.Experiments = new Api.ExperimentsApi(apiConfig);
+  detApi.StreamingExperiments = Api.ExperimentsApiFetchParamCreator(apiConfig);
```

Comment:
**nit suggestion:** a slightly different approach to avoid modifying incoming param
```
export const updateDetApi = (config: Api.Configuration): void => {
  const apiConfig = {
    apiKey: 'Bearer ' + globalStorage.get(STORAGE_KEY_AUTH),
    basePath: serverAddress(),
    ...config,
  };
  ...
};
```